### PR TITLE
Display avatars in user search

### PR DESCRIPTION
**Pull request by tomasr8**
When searching users, instead of the default icon, avatars are displayed.
If the user does not have an avatar, the default avatar with the first letter of
the first name is used.

The group search still displays the default result icon.

Before:
Screenshot from 2021-06-08 16-41-21
After:
Screenshot from 2021-06-08 18-07-55
Screenshot from 2021-06-09 10-47-43

### CODE REVIEW

File: indico/modules/core/controllers.py [hunk 1]
```diff
@@ -199,6 +200,8 @@ def _serialize_principal(self, identifier, principal):
                     'last_name': principal.last_name,
                     'email': principal.email,
                     'affiliation': principal.affiliation,
+                    'avatar_url': (principal.avatar_url
+                                   if principal.avatar_url else get_avatar_from_name(principal.first_name)),
```
Comment (ThiefMaster) [inline indico/modules/core/controllers.py:204]: I don't think the if/else is needed here; users always have a suitable `avatar_url`

File: indico/modules/users/util.py [hunk 2]
```diff
@@ -441,3 +441,8 @@ def send_avatar(user):
     return send_file('avatar.png', BytesIO(user.picture), mimetype=metadata['content_type'],
                      inline=True, conditional=True, last_modified=parse_date(metadata['lastmod']),
                      cache_timeout=(86400*7))
+
+
+def get_avatar_from_name(first_name):
```
Comment (ThiefMaster) [inline indico/modules/users/util.py:446]:
```suggestion
def get_avatar_url_from_name(first_name):
```

that name is clearer IMO (don't forget to chance the places where it's called/imported as well ofc)

File: indico/modules/users/controllers.py [hunk 3]
```diff
@@ -696,6 +696,8 @@ def _serialize_pending_user(self, entry):
         affiliation = entry.data.get('affiliation') or ''
         email = entry.data['email'].lower()
         ext_id = f'{entry.provider.name}:{entry.identifier}'
+        avatar_url = entry.data.get('avatar_url') or get_avatar_from_name(first_name)
```
Comment (ThiefMaster) [inline indico/modules/users/controllers.py:699]: AFAIL `entry` here is an `IdentityInfo` object from flask-multipass which never has an avatar URL, so you can always call the util here

File: indico/web/client/js/react/components/principals/Search.jsx [hunk 4]
```diff
@@ -58,38 +59,48 @@ const searchFactory = config => {
...
-            </List.Description>
-          )}
+  const FavoriteItem = ({name, detail, added, onAdd, onRemove, avatarURL}) => {
+    const avatar = avatarURL ? (
+      <Image src={avatarURL} size="mini" avatar />
+    ) : (
+      <Icon name={resultIcon} />
+    );
+
+    const marginLeft = avatarURL ? 'margin-left' : '';
```
Comment (ThiefMaster) [inline indico/web/client/js/react/components/principals/Search.jsx:69]: I don't think that's a good name for a CSS class. I'd try to find a name that describes the meaning, e.g. `has-avatar`.

File: indico/web/client/js/react/components/principals/Search.jsx [hunk 5]
```diff
@@ -141,6 +152,7 @@ const searchFactory = config => {
                         onRemove={() => {
                           onRemove(x);
                         }}
+                        avatarURL={x.avatarURL}
```
Comment (ThiefMaster) [inline indico/web/client/js/react/components/principals/Search.jsx:155]: I'd move it above the callbacks ;) looks cleaner IMO if the single-line props come first

Comment (tomasr8) [reply indico/web/client/js/react/components/principals/Search.jsx]: Fixed ;)

File: indico/web/client/js/react/components/principals/Search.jsx (hunk 4 above)
Comment (tomasr8) [reply indico/web/client/js/react/components/principals/Search.jsx]: I used `has-avatar`

File: indico/modules/users/controllers.py (hunk 3 above)
Comment (tomasr8) [reply indico/modules/users/controllers.py]: Removed ;)

File: indico/modules/users/util.py (hunk 2 above)
Comment (tomasr8) [reply indico/modules/users/util.py]: Renamed the function

File: indico/modules/core/controllers.py (hunk 1 above)
Comment (tomasr8) [reply indico/modules/core/controllers.py]: Removed the redundant check ;)

File: indico/web/client/js/react/components/principals/Search.jsx [hunk 6]
```diff
@@ -58,38 +59,48 @@ const searchFactory = config => {
...
-            </List.Description>
-          )}
+  const FavoriteItem = ({name, detail, added, onAdd, onRemove, avatarURL}) => {
+    const avatar = avatarURL ? (
+      <Image src={avatarURL} size="mini" avatar />
+    ) : (
+      <Icon name={resultIcon} />
+    );
+
+    const hasAvatar = avatarURL ? 'has-avatar' : '';
```
Comment (ThiefMaster) [inline indico/web/client/js/react/components/principals/Search.jsx:69]: `hasAvatarClass`? the current name sounds very bool-ish

**Review by ThiefMaster — APPROVED** [review]
Tested it locally and it works great! :) Just one more small naming suggestion...

File: indico/web/client/js/react/components/principals/Search.jsx (hunk 6 above)
Comment (tomasr8) [reply indico/web/client/js/react/components/principals/Search.jsx]: You're right, renamed ;)

**Pull request merged**